crates/sparrow-sources/src/in_memory.rs

```diff
@@ -0,0 +1,221 @@
+use std::sync::{Arc, RwLock};
+
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+use error_stack::{IntoReportCompat, ResultExt};
+use futures::{Stream, StreamExt, TryStreamExt};
+
+use sparrow_batch::Batch;
+use sparrow_core::ReadConfig;
```

Comment:
Another point -- this only depends on `sparrow_core` for `ReadConfig`. If that was in `sparrow-interfaces` we wouldn't need to depend on it.